java/configuration/src/main/java/sleeper/configuration/properties/InstanceProperties.java

```diff
@@ -137,4 +142,11 @@ public static String tagsToString(Map<String, String> tags) {
         return builder.toString();
     }
 
+    public List<InstanceProperty> getAllProperties() {
+        List<InstanceProperty> userProperties = Arrays.asList(UserDefinedInstanceProperty.values());
+        List<InstanceProperty> systemProperties = Arrays.asList(SystemDefinedInstanceProperty.values());
+        return Stream.concat(userProperties.stream(), systemProperties.stream())
+                .sorted(Comparator.comparing(property -> PropertyGroup.all().indexOf(property.getPropertyGroup())))
```

Comment:
It looks like this sort is used to combine the user and system properties into a single list but organised by the property group?

Please can we unit test this in this module? We could extract a function to combine two lists of properties, and test against that so that we can separate out the scenarios rather than use the full lists in every test.

In particular, it seems important to test that the ordering in the original lists is preserved and that they're combined in a stable way, eg. user defined followed by system defined.

We could name this method to describe the fact that it's organising them by their groups, or maybe return a map of property group to list of properties.

Reply:
I'm not actually clear on whether there should be some alphabetical sorting at some point.